proxy_agent/src/key_keeper.rs

```diff
@@ -97,18 +98,18 @@ fn poll_secure_channel_status(
         if !first_iteration {
             // skip the sleep for the first loop
 
-            let sleep =
-                if key_keeper_wrapper::get_current_secure_channel_state(shared_state.clone())
+            let sleep = if key_keeper_wrapper::get_current_secure_channel_state(shared_state.clone())
                     == UNKNOWN_STATE
+                    && FREQUENT_PULL_INTERVAL < interval        // test internal is less than 1 second
                     && helpers::get_elapsed_time_in_millisec()
                         < FREQUENT_PULL_TIMEOUT_IN_MILLISECONDS
-                {
-                    // frequent poll the secure channel status every second for the first 5 minutes
-                    // until the secure channel state is known
-                    FREQUENT_PULL_INTERVAL
-                } else {
-                    interval
-                };
+            {
+                // frequent poll the secure channel status every second for the first 5 minutes
+                // until the secure channel state is known
+                FREQUENT_PULL_INTERVAL
+            } else {
+                interval
+            };
             thread::sleep(sleep);
```

Comment:
With the new concurrent model, does `thread::sleep()` prevent other things from happening inside the main thread? Should we use tokios builtin [sleep](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html) function instead, or does it not matter here?